build.gradle

```diff
@@ -16,22 +16,25 @@ repositories {
 }
 
 dependencies {
-    implementation 'commons-cli:commons-cli:1.5.0'
-    implementation 'org.apache.commons:commons-csv:1.10.0'
-    implementation 'org.apache.poi:poi-ooxml:5.2.3'
-    implementation 'com.fasterxml.jackson.core:jackson-core:2.15.2'
-    implementation 'com.fasterxml.jackson.core:jackson-annotations:2.15.2'
-    implementation 'com.fasterxml.jackson.core:jackson-databind:2.15.2'
-    implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.15.2'
-    implementation "com.github.spotbugs:spotbugs:4.8.3"
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.9.3'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.9.3'
+    implementation 'commons-cli:commons-cli:1.9.0'
+    implementation 'org.apache.commons:commons-csv:1.14.0'
+    implementation 'org.apache.poi:poi-ooxml:5.4.1'
+    implementation 'com.fasterxml.jackson.core:jackson-core:2.19.0'
+    implementation 'com.fasterxml.jackson.core:jackson-annotations:2.19.0'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.19.0'
+    implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.19.0'
+    implementation "com.github.spotbugs:spotbugs:4.9.3"
+
+    testImplementation platform('org.junit:junit-bom:5.13.0-M2')
```

Comment:
I would stick with 5.12.2 for Jupiter to avoid `-M*`. See if that allows you to keep the old format we've been using consistently in this file for a few years.